### PR TITLE
Fixes for validation process with GCP and Azure

### PR DESCRIPTION
**Pull request by malscent**

### CODE REVIEW

File: .github/workflows/Azure-CreateRelease.yml [hunk 1]
```diff
@@ -50,34 +50,34 @@ jobs:
         run: |
           bash ${GITHUB_WORKSPACE}/azure/CouchbaseServer-Application/makeArchives.sh \
                                           -l byol \
-                                          -s byol_2019 \
+                                          -s cbs-2022-byol-pre-installed \
                                           -p couchbase \
                                           -o couchbase-server-enterprise \
-                                          -v 18.4.0
+                                          -v latest
```
Comment (tahmmee) [inline .github/workflows/Azure-CreateRelease.yml]: 'latest' seems like a risk between deployments if tag secretly points to new version  (and elsewhere below)

File: azure/Couchbase-VirtualMachine/gateway-startup.sh [hunk 2]
```diff
@@ -146,5 +197,27 @@ if [[ "$RUNNING" != "200" ]]; then
     echo "Sync Gateway is not running.  We should restart."
     service sync_gateway stop &> /dev/null
     service sync_gateway start &> /dev/null
+    sleep 10
 fi
 
+RUNNING=$(curl -s -o /dev/null -I -w "%{http_code}" http://localhost:4984)
+if [[ "$RUNNING" == "200" && "$VERSION" =~ ^3 ]]; then
```
Comment (tahmmee) [inline azure/Couchbase-VirtualMachine/gateway-startup.sh:204]: SG should definitely be running at this point.. but in the chance it isn't, I think printing out a message about "service did not start" will help debugging.

File: azure/Couchbase-VirtualMachine/gateway-startup.sh [hunk 3]
```diff
@@ -146,5 +197,27 @@ if [[ "$RUNNING" != "200" ]]; then
...
+  # here we need to hit the API and configure the database for sync gateway 3.0+
+  CONFIGURED=$(curl -L -s -o /dev/null -I -w "%{http_code}"  http://127.0.0.1:4985/$BUCKET/_config -X GET --user $USERNAME:$PASSWORD)
+  if [[ "$CONFIGURED" != "200" ]]; then
+    RESPONSE=$(curl -X PUT "http://127.0.0.1:4985/$BUCKET/" \
+        -H "accept: application/json" \
+        -H "Content-Type: application/json" \
+        -d "{\"bucket\": \"$BUCKET\",\"name\": \"$BUCKET\", \"num_index_replicas\":0}" \
+        --user $USERNAME:$PASSWORD -L -s -w '%{http_code}')
+    while [[ "$RESPONSE" != "412" ]];
```
Comment (tahmmee) [inline azure/Couchbase-VirtualMachine/gateway-startup.sh:213]: risking an infinite loop if the PUT config is failing for irrecoverable reasons. perhaps a retry count is in order.

File: azure/Couchbase-VirtualMachine/gateway-startup.sh (hunk 2 above)
Comment (malscent) [reply azure/Couchbase-VirtualMachine/gateway-startup.sh:204]: Agreed.. Fixing

File: azure/Couchbase-VirtualMachine/gateway-startup.sh (hunk 3 above)
Comment (malscent) [reply azure/Couchbase-VirtualMachine/gateway-startup.sh:213]: Agreed.. Fixing

**Pull request merged**